metaflow/plugins/tagging_cli.py

```diff
@@ -0,0 +1,424 @@
+            all_runs.extend([' ' for _ in range(addl_runs)])
+
+        lines = []
+        lines.append('For tag %s' % (cat if is_system else '*%s*' % cat))
+        for i in range(words_per_column):
+            line_values = [all_runs[i+j*words_per_column] for j in range(column_count)]
+            length_values = [max_length for l in line_values]
+            formatter = "    ".join(["{:<%d}" % l for l in length_values])
+            lines.append(formatter.format(*line_values))
+        lines.append("")
+        obj.echo('\n'.join(lines), err=False)
+    else:
+        # In quiet mode, we display things much more simply so it is parseable.
+        # When displaying by tag, we output twp columns:
+        #  - the tag (user:<tag> or sys:<tag>)
+        #  - the comma separated list of runs
+        # Columns are separated by a semicolon surrounded by a space on each side
+        # to make it consistent with the by_run listing
+        obj.echo_always(' ; '.join([
+            'system:%s' % cat if is_system else 'user:%s' % cat,
+            ','.join(runs)]))
+
+
+def print_tags(obj, metaflow_run_client):
+    system_tags = {metaflow_run_client.pathspec: metaflow_run_client.system_tags}
+    all_tags = {metaflow_run_client.pathspec: metaflow_run_client.tags}
+    return format_tags(obj, system_tags, all_tags, by_tag=False)
+
+
+def is_prod_token(token):
+    return token is not None and token.startswith("production:")
+
+
+def obtain_metaflow_run_client(obj, run_id, user_namespace):
+    if is_prod_token(user_namespace):
+        raise CommandException(
+            "Modifying a scheduled run's tags is currently not allowed.")
+
+    flow_name = obj.flow.name
+
+    # handle error messaging for two cases
+    # 1. our user tries to tagging a new flow before it is run
+    # 2. our user makes a typo in --namespace
+    try:
+        namespace(user_namespace)
+        Flow(pathspec=flow_name)
+    except MetaflowNotFound:
+        raise CommandException("Flow *%s* does not exist " % flow_name +
+                               "in any namespace.\n" +
+                               "Please run the flow before tagging.")
+    except MetaflowNamespaceMismatch:
+        raise CommandException("Flow *%s* does not belong to " % flow_name +
+                               "namespace *%s*.\n" % user_namespace +
+                               "Check typos if you used option --namespace.")
+
+    # throw an error with message to include latest run-id when run_id is None
+    if run_id is None:
+        latest_run_id = Flow(pathspec=flow_name).latest_run.id
+        msg = "Please specify a run-id using --run-id\n" + \
+              "Flow %s's latest run has run-id *%s* under namespace %s" \
+            % (flow_name, latest_run_id, user_namespace)
+        raise CommandException(msg)
+    run_id_parts = run_id.split('/')
+    if len(run_id_parts) == 1:
+        run_name = "%s/%s" % (flow_name, run_id)
+    elif len(run_id_parts) == 2:
```

Comment:
Why do we have this check? We should expect the run-id to be a run-id. That is the behavior elsewhere with other CLIs as well in Metaflow.

Reply:
This is because you can do `tag list MyFlow/RunID`. Am fine not doing that if you don't want to though. I'll just have to check if it breaks something internally.

Reply:
removed and just kept `run_id`.